src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/SymbolAnalysisContextExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles
+{
+    internal static class SymbolAnalysisContextExtensions
+    {
+        public static async Task<NamingStylePreferencesInfo> GetNamingStylePreferencesAsync(this SymbolAnalysisContext context)
+        {
+            var location = context.Symbol.Locations.FirstOrDefault();
+            if (location == null)
+            {
+                return null;
+            }
+
+            var cancellationToken = context.CancellationToken;
+            var syntaxTree = location.SourceTree;
+            var options = context.Options;
+            var optionSet = await options.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).ConfigureAwait(false);
+            string language = context.Compilation.Language;
+            var currentValue = optionSet.GetOption(SimplificationOptions.NamingPreferences, language);
+            int hashCode = currentValue.GetHashCode();
+            var viewModel = SerializableNamingStylePreferencesInfo.FromXElement(XElement.Parse(currentValue));
+            var preferences = viewModel.GetPreferencesInfo();
+            return preferences;
```

Comment:
It terrifies me that we don't do any caching here and instead populate all the rules from editorconfig, serialize them to xml and then deserialize back from xml ON EVERY SYMBOL?